### PR TITLE
Reduce duplicate code

### PR DESCRIPTION
**Pull request by Utesgui**
And add generic functions/scripts

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -8,9 +8,20 @@ jobs:
...
+          uses: actions/checkout@v4
+        - name: Update package
+          id: update_version
+          env:
+            GITHUB_TOKEN: ${{ secrets.WINGET_PAT }}
+            WINGET_PKGS_FORK_REPO: ${{ vars.WINGET_PKGS_FORK_REPO }}
+            WebsiteURL: "https://www.isobuster.com/downloads/isobuster/isobuster_install_64bit.exe https://www.isobuster.com/downloads/isobuster/isobuster_install.exe"
+            PackageName: "SmartProjects.IsoBuster"
+            With: "Komac"
+          run: .\Scripts\generic.ps1
```
Comment (a-mnich) [inline .github/workflows/test.yml:27]: even though the original flow is unused, we don't need to merge this test :D

File: Scripts/common.ps1 [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    $progressPreference = 'silentlyContinue'
+    $latestWingetMsixBundleUri = $(Invoke-RestMethod https://api.github.com/repos/microsoft/winget-cli/releases/latest).assets.browser_download_url | Where-Object { $_.EndsWith(".msixbundle") }
+    $latestWingetMsixBundle = $latestWingetMsixBundleUri.Split("/")[-1]
+    Write-Host "Downloading winget to artifacts directory..."
+    Invoke-WebRequest -Uri $latestWingetMsixBundleUri -OutFile "./$latestWingetMsixBundle"
+    Invoke-WebRequest -Uri https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx -OutFile Microsoft.VCLibs.x64.14.00.Desktop.appx
+    Add-AppxPackage Microsoft.VCLibs.x64.14.00.Desktop.appx
+    Add-AppxPackage $latestWingetMsixBundle
+
+    start-sleep 20
```
Comment (a-mnich) [inline Scripts/common.ps1:55]: extract winget install in it's own function and ideally also check if install is even necessarry (or if winget is already installed)

File: Scripts/common.ps1 [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+    $foundMessage, $textVersion, $separator, $wingetVersions = winget search --id $wingetPackage --source winget --versions
+
+    if (!$wingetVersions) {
+        Write-Host "Packet not yet in winget. Please add new Packet manually"
```
Comment (a-mnich) [inline Scripts/common.ps1:60]:
```suggestion
        Write-Host "Package not yet in winget. Please add new package manually"
```

File: Scripts/common.ps1 [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+#     $Latest = Get-VersionAndUrl -wingetPackage $wingetPackage -WebsiteURL $url
+
+#     Update-WingetPackage -wingetPackage $wingetPackage -latestVersion $Latest.Version -with Komac -latestVersionUrls $Latest.URLs
+# }
+
+
+
+$wingetPackage = ${Env:PackageName}
+$WebsiteURL = ${Env:WebsiteURL}
+$githubToken
```
Comment (a-mnich) [inline Scripts/common.ps1:186]:
```suggestion
```

File: Scripts/common.ps1 [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+
+    if ($PackageAndVersionInWinget) {
+
+        $ExistingPRs = Test-ExistingPRs -wingetPackage $wingetPackage -latestVersion $($Latest.Version)
+        
+        if ($ExistingPRs) {
+            Write-Host "Downloading $with and open PR for $wingetPackage Version $($Latest.Version)"
+            Switch ($with) {
+                "Komac" {
+                    Invoke-WebRequest "https://github.com/russellbanks/Komac/releases/download/v2.2.1/KomacPortable-x64.exe" -OutFile komac.exe
```
Comment (a-mnich) [inline Scripts/common.ps1:160]: We should either make the komac version configurable or use the latest.  
As we already use latest for wingetcreate I would prefer the same for komac

File: Scripts/generic.ps1 [hunk 6]
```diff
@@ -0,0 +1,12 @@
+. .\Scripts\common.ps1
+
+#### Main
+
+if (${Env:With}) {
+    Update-WingetPackage -wingetPackage ${Env:PackageName} -With ${Env:With} -WebsiteURL ${Env:WebsiteURL}
+}
+else {
+    Update-WingetPackage -wingetPackage ${Env:PackageName} -WebsiteURL ${Env:WebsiteURL}
+}
```
Comment (a-mnich) [inline Scripts/generic.ps1:10]: Empty "-With" should evaluate to $null and therefore "komac" should be used as default.  
-> no if clause necesarry

**Review by a-mnich — CHANGES_REQUESTED** [review]

File: Scripts/common.ps1 (hunk 5 above)
Comment (Utesgui) [reply Scripts/common.ps1:160]: ja dann mach doch mein Gott

File: Scripts/generic.ps1 (hunk 6 above)
Comment (Utesgui) [reply Scripts/generic.ps1:10]: cool, thought the same however "" is not a valid case in the validation set. So please fix it with a working solution thank you.

File: Scripts/common.ps1 (hunk 2 above)
Comment (Utesgui) [reply Scripts/common.ps1:55]: should not be used anyway ... just there for future switch back

**Pull request merged**